pkg/publicapi/middleware/version.go

```diff
@@ -29,11 +29,20 @@ type VersionCheckError struct {
 	ServerVersion string
 }
 
+func stripPreRelease(version *semver.Version) (*semver.Version, error) {
+	newVersion, _ := semver.NewVersion(version.String())
+	_, err := newVersion.SetPrerelease("")
+	if err != nil {
+		return nil, err
+	}
+	return newVersion, nil
+}
+
```

Comment:
```suggestion
func stripPreRelease(versionStr string) (*semver.Version, error) {
	v, err := semver.NewVersion(versionStr)
	if err != nil {
		return version.Unknown, fmt.Errorf("failed to parse client version: %w", err)
	}
	strippedVersion, err := v.SetPrerelease("")
	if err != nil {
		return version.Unknown, fmt.Errorf("failed to strip pre-release version from client: %w", err)
	}
	return &strippedVersion, nil
}
```
please move this utility method to the bottom of the file.